src/plone/recipe/zope2instance/recipe.py

```diff
@@ -296,7 +296,7 @@ def build_zope_conf(self):
             chameleon_cache = 'CHAMELEON_CACHE {}'.format(chameleon_cache)
             if environment_vars and '\n' in environment_vars:
                 # default case
-                environment_vars += '\n'.format(chameleon_cache)
+                environment_vars += '\n'
```

Comment:
But then `chameleon_cache` is not added. I think this was originally meant to be:

    environment_vars += '\n{}'.format(chameleon_cache)

If you prefer, it could be done differently to avoid repeating the `chameleon_cache` so often, something like:

```
if environment_vars:
    if '\n in environment_vars:
        environment_vars += '\n'
    else:
        environment_vars += ' '
environment_vars += chameleon_cache
```

Maybe the case that `environment_vars` is None should be checked.

Reply:
I picked the first cause it changes less lines.
IMHO it would be nice to factor out the code that computes the environment_vars to a method to make the things more manageable and easier to test.
But that goes beyond the issue I would like to fix :)